backend/de.metas.business/src/main/java/de/metas/invoice/invoiceProcessingServiceCompany/InvoiceProcessingServiceCompanyService.java

```diff
@@ -93,11 +99,12 @@ public InvoiceProcessingServiceCompanyService(
 
 		final ImmutableSet<InvoiceId> invoiceIdsWithGeneratedFees = retainIfServiceInvoiceWasAlreadyGenerated(ImmutableSet.of(invoiceId));
 
-		if (invoiceIdsWithGeneratedFees.size() > 0)
+		if (!invoiceIdsWithGeneratedFees.isEmpty())
 		{
-			throw new AdempiereException("Service Invoice already generated")
-					.setParameter("invoiceId", invoiceId)
-					.appendParametersToMessage();
+			final String documentNo = invoiceDAO.getDocumentNosByInvoiceIds(ImmutableSet.of(invoiceId)).get(invoiceId);
+
+			final ITranslatableString errorMsg = msgBL.getTranslatableMsgText(MSG_INVOICE_HAS_SERVICE_INVOICE, documentNo);
+			throw new AdempiereException(errorMsg);
```

Comment:
u can pass the MSG_INVOICE_HAS_SERVICE_INVOICE directly. don't have to convert it to ITranslatableString.
see org.adempiere.exceptions.AdempiereException#AdempiereException(de.metas.i18n.AdMessageKey, java.lang.Object...)

Reply:
done, thx